apps/storefront/src/components/HeadlessController.tsx

```diff
@@ -186,6 +191,20 @@ export default function HeadlessController({ setOpenPage }: HeadlessControllerPr
             storeDispatch(setB2BToken(b2bStorefrontJWTToken));
             await getCurrentCustomerInfo(b2bStorefrontJWTToken);
           },
+          logout: async () => {
+            try {
+              if (isAgenting) {
+                await endMasquerade();
+              }
+            } catch (e) {
+              b2bLogger.error(e);
+            } finally {
+              // SUP-1282 Clear sessionStorage to allow visitors to display the checkout page
+              window.sessionStorage.clear();
+              logoutSession();
+              window.b2b.callbacks.dispatchEvent('on-logout' as any);
+            }
```

Comment:
we should avoid using `as any` - this one already has an Enum we can use.
```suggestion
              window.b2b.callbacks.dispatchEvent(B2BEvent.OnLogout);
```